File: PythonAbril30dias/dia7.py
import random

snail_a = []
snail_b = []
snail_a_name = "A"
snail_b_name = "B"

def get_snail_results(snail):
    max_distance = 20
    runned_distance = 0
    while max_distance > runned_distance:
        # Determinar velocidad y longitud recorrida del caracol
        speed = float(random.randint(100, 500) / 100)
        runned_distance += speed
        runned_distance = 20.00 if runned_distance > 20 else runned_distance
        snail.append(runned_distance)
    return snail

def format_results(snail, snail_name):
# Imprimir y formatear recorrido del caracol
    print(f"Caracol {snail_name}]: ", end="\t")
    for i in snail:
        print(f"{i:6.2f} m.", end="\t")
    print()
    return snail

snail_a = get_snail_results(snail_a)
snail_b = get_snail_results(snail_b)

format_results(snail_a, snail_a_name)
format_results(snail_b, snail_b_name)

if len(snail_a) < len(snail_b):
    print(f"¡¡¡Ha ganado el caracol {snail_a_name}!!!")
elif len(snail_a) > len(snail_b):
    print(f"¡¡¡Ha ganado el caracol {snail_b_name}!!!")
else:
    print("¡¡¡Los caracoles han empatado!!!")

File: JavaAbril30dias/src/Dia7.java
public class Dia7 {
    public static void main (String[]args) {
        double maxDistance = 20;
        double runnedDistance = 1;
        int snailA = printRaceResults("A");
        int snailB = printRaceResults("B");
        if (snailB > snailA) {
            System.out.println("El ganador es el A");
        }
        else {
            System.out.println("El ganador es B");
        }
    }
    public static int printRaceResults (String name ) {
        double maxDistance = 20;
        double runnedDistance = 1;
        int iterations = 0;
        while (maxDistance > runnedDistance) {
            double speed = (Math.random()*4) + 1;
            runnedDistance += speed;
            runnedDistance = Math.round(runnedDistance * 100.0) / 100.0;
            runnedDistance = runnedDistance > 20 ? 20.00 : runnedDistance;
            System.out.print("Caracol " + name + ":\t");
            System.out.print(runnedDistance + " m.\t");
            iterations++;
        }
        System.out.println();
        return iterations;
    }
}
